41-vaccine-temp-monitor/firmware/arduino/src/main.cpp

```diff
@@ -0,0 +1,528 @@
+        {
+            if (J *aux1 = JGetArrayItem(state, 0))
+            {
+                result = JGetBool(aux1, "low");
+            }
+        }
+    }
+
+    // Clean up and exit
+    notecard.deleteResponse(rsp);
+    return result;
+}
+
+void setup()
+{
+    // Provide visual signal when the Host MCU is powered
+    ::pinMode(LED_BUILTIN, OUTPUT);
+    ::digitalWrite(LED_BUILTIN, HIGH);
+
+#ifndef NDEBUG
+#warning "Debug mode is enabled. Define NDEBUG to disable debug."
+    // Initialize Debug Output
+    usbSerial.begin(115200);
+    static const size_t MAX_SERIAL_WAIT_MS = 5000;
+    size_t begin_serial_wait_ms = ::millis();
+    while (!usbSerial && (MAX_SERIAL_WAIT_MS > (::millis() - begin_serial_wait_ms)))
+    {
+        ; // wait for debug serial port to connect. Needed for native USB
+    }
+    notecard.setDebugOutputStream(usbSerial);
+#endif
+
+    // Initialize LM75A
+    lm75a.begin();
+
+    // Initialize Notecard
+    notecard.begin();
+#ifdef ARDUINO_ARCH_ESP32
+    // Initialize Auxiliary Wi-Fi
+    aux_wifi.begin();
+#endif
+
+    // Configure Notecard
+    configureNotecard();
+
+    // Fetch Environment Variables
+    fetchEnvironmentVariables();
+
+    // Configure temperature threshold for Controlled Room Temperature (CRT)
+    // U.S. Pharmacopeia (USP) USP <659> "Packaging and Storage Requirements"
+    //
+    // Controlled room temperature: The temperature maintained thermostatically
+    // that encompasses at the usual and customary working environment of 20°-25°
+    // (68°-77° F). Excursions between 15° and 30° (59° and 86° F) that are
+    // experienced in pharmacies, hospitals, and warehouses, and during shipping
+    // are allowed. Provided the mean kinetic temperature does not exceed 25°,
+    // transient spikes up to 40° are permitted as long as they do not exceed 24h.
+    // Spikes above 40° may be permitted only if the manufacturer so instructs.
+    // https://www.gmp-compliance.org/gmp-news/what-are-the-regulatory-definitions-for-ambient-room-temperature-and-cold-chain
+    lm75a.setOsTripTemperature(alarm_temperature);    // Max safe temperature for travel
+    lm75a.setHysterisisTemperature(alarm_hysteresis); // Max safe temperature for storage
+
+    // Sample temperature sensor
+    const float temperature = lm75a.getTemperature();
+    static const bool ACTIVE_ALARM = (temperature > alarm_temperature);
+    static const bool TEMP_WARNING = inHysteresis(); // Read AUX1 pin to determine
+                                                     // if we are in hysteresis
+
+    // Sample battery voltage
+    float voltage = 0.0f;
+    bool low_battery = false;
+    if (J *rsp = notecard.requestAndResponse(notecard.newRequest("card.voltage")))
+    {
+        voltage = JGetNumber(rsp, "value");
+        const char *power_mode = JGetString(rsp, "mode");
+        low_battery = (power_mode && !strcmp(power_mode, "low"));
+        NoteDeleteResponse(rsp);

```

Comment:
We're using both NoteDeleteResponse and notecard.deleteResponse in this code. Probably best to settle on one or the other.

Reply:
`NoteDeleteResponse()` snuck in there. It's dead now :skull: